google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/BigtableChannelPrimer.java

```diff
@@ -141,41 +86,21 @@ private void sendPrimeRequests(ManagedChannel managedChannel) throws IOException
             .build();
 
     try (EnhancedBigtableStub stub = EnhancedBigtableStub.create(primingSettings)) {
-      Map<String, ApiFuture<?>> primeFutures = new HashMap<>();
-
-      // Prime all of the table ids in parallel
-      for (String tableId : tableIds) {
-        ApiFuture<Row> f =
-            stub.createReadRowsRawCallable(new DefaultRowAdapter())
-                .first()
-                .futureCall(
-                    ReadRowsRequest.newBuilder()
-                        .setTableName(
-                            TableName.format(
-                                primingSettings.getProjectId(),
-                                primingSettings.getInstanceId(),
-                                tableId))
-                        .setAppProfileId(primingSettings.getAppProfileId())
-                        .setRows(RowSet.newBuilder().addRowKeys(PRIMING_ROW_KEY).build())
-                        .setFilter(RowFilter.newBuilder().setBlockAllFilter(true).build())
-                        .setRowsLimit(1)
-                        .build());
+      PingAndWarmRequest request =
+          PingAndWarmRequest.newBuilder()
+              .setName(
+                  NameUtil.formatInstanceName(
+                      primingSettings.getProjectId(), primingSettings.getInstanceId()))
+              .setAppProfileId(primingSettings.getAppProfileId())
+              .build();
 
-        primeFutures.put(tableId, f);
-      }
-
-      // Wait for all of the prime requests to complete.
-      for (Map.Entry<String, ApiFuture<?>> entry : primeFutures.entrySet()) {
-        try {
-          entry.getValue().get();
-        } catch (Throwable e) {
-          if (e instanceof ExecutionException) {
-            e = e.getCause();
-          }
-          LOG.warning(
-              String.format(
-                  "Failed to prime channel for table: %s: %s", entry.getKey(), e.getMessage()));
+      try {
+        stub.pingAndWarmCallable().call(request);
+      } catch (Throwable e) {
```

Comment:
Add a TODO here: I'm not sure that we should swallow the error here. We are pre-emptively swapping channels and the new channel is bad, we might want to consider on swapping it. For now leave it as is